Gardevoir tracing Dancer

When Gardevoir has the Trace Ability and the opponent's Oricorio with the ability Dancer uses a move that is affected by Dancer, an `AssertionError` occurs:

`AssertionError: Error with move psyshock. Expected self.moves to contain copycat, metronome, mefirst, mirrormove, assist, transform or mimic. Got {'quiverdance': quiverdance (Move object), 'revelationdance': revelationdance (Move object), 'moonblast': moonblast (Move object), 'mysticalfire': mysticalfire (Move object)}
`

This happens because there are no fallbacks for cases where the Pokémon has moved with its ability. This could be fixed by checking if the ability is Trace or Dancer (maybe Magic Bounce as well).

--- a/src/poke_env/battle/pokemon.py
+++ b/src/poke_env/battle/pokemon.py
@@ -661,7 +661,7 @@ class Pokemon:
                     "transform",
                     "mimic",
                 }.intersection(self.moves)
-                
+
                 """
                 Check if the pokemon has abilities that can grant moves
 
@@ -671,7 +671,7 @@ class Pokemon:
                 has_move_granting_ability = (
                     self.ability in ("dancer", "trace") if self.ability else False
                 )
-                
+
                 if not has_copy_move and not has_move_granting_ability:
                     assert False, (
                         f"Error with move {move}. Expected self.moves to contain copycat, "

